client/simple/src/js/main/results.ts

```diff
@@ -0,0 +1,178 @@
+  const btnLabelCollapsed = this.getAttribute("data-btn-text-collapsed");
+  const btnLabelNotCollapsed = this.getAttribute("data-btn-text-not-collapsed");
+  const target = this.getAttribute("data-target");
+
+  if (!(target && btnLabelCollapsed && btnLabelNotCollapsed)) return;
+
+  const targetElement = document.querySelector<HTMLElement>(target);
+  assertElement(targetElement);
+
+  const isCollapsed = this.classList.contains("collapsed");
+  const newLabel = isCollapsed ? btnLabelNotCollapsed : btnLabelCollapsed;
+  const oldLabel = isCollapsed ? btnLabelCollapsed : btnLabelNotCollapsed;
+
+  this.innerHTML = this.innerHTML.replace(oldLabel, newLabel);
+  this.classList.toggle("collapsed");
+
+  targetElement.classList.toggle("invisible");
+});
+
+listen("click", ".media-loader", function (this: HTMLElement) {
+  const target = this.getAttribute("data-target");
+  if (!target) return;
+
+  const iframeLoad = document.querySelector<HTMLIFrameElement>(`${target} > iframe`);
+  assertElement(iframeLoad);
+
+  const srctest = iframeLoad.getAttribute("src");
+  if (!srctest) {
+    const dataSrc = iframeLoad.getAttribute("data-src");
+    if (dataSrc) {
+      iframeLoad.setAttribute("src", dataSrc);
+    }
+  }
+});
+
+listen("click", "#copy_url", async function (this: HTMLElement) {
+  const target = this.parentElement?.querySelector<HTMLPreElement>("pre");
+  assertElement(target);
+
+  await navigator.clipboard.writeText(target.innerText);
+  const copiedText = this.dataset.copiedText;
+  if (copiedText) {
+    this.innerText = copiedText;
+  }
+});
+
+listen("click", ".result-detail-close", (event: Event) => {
+  event.preventDefault();
+  mutable.closeDetail?.();
+});
+
+listen("click", ".result-detail-previous", (event: Event) => {
+  event.preventDefault();
+  mutable.selectPrevious?.(false);
+});
+
+listen("click", ".result-detail-next", (event: Event) => {
+  event.preventDefault();
+  mutable.selectNext?.(false);
+});
+
+// listen for the back button to be pressed and dismiss the image details when called
+window.addEventListener("hashchange", () => {
+  if (window.location.hash !== "#image-viewer") {
+    mutable.closeDetail?.();
+  }
+});
+
+const swipeHorizontal: NodeListOf<HTMLElement> = document.querySelectorAll<HTMLElement>(".swipe-horizontal");
+for (const element of swipeHorizontal) {
+  listen("swiped-left", element, () => {
+    mutable.selectNext?.(false);
+  });
+
+  listen("swiped-right", element, () => {
+    mutable.selectPrevious?.(false);
+  });
+}
+
+window.addEventListener(
+  "scroll",
+  () => {
+    const backToTopElement = document.getElementById("backToTop");
+    const resultsElement = document.getElementById("results");
+
+    if (backToTopElement && resultsElement) {
+      const scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
+      const isScrolling = scrollTop >= 100;
+      resultsElement.classList.toggle("scrolling", isScrolling);
```

Comment:
Not related to this PR (same issue in current master branch):  When [`infinite_scroll:`](https://docs.searxng.org/admin/settings/settings_ui.html) is active, the *back to top* button is never shown  / especially in *infinitely long* result lists such a button would be convenient.